src/views/Repay/RepayDetail.tsx

```diff
@@ -96,7 +97,11 @@ const InitialComp: React.FC<{
     }
 
     // availableToRepay = min(debt, balance)
-    return userBalance.gt(debtForAsset) ? debtForAsset : userBalance;
+    // 10% gap to make sure the interest accrued during the time between fetching and repaying
+    // is accounted for ->  (difference gets minted as agToken)
+    return userBalance.gt(debtForAsset.mul(11).div(10))
+      ? debtForAsset.mul(10).div(9).add(MINIMUM_NATIVE_RESERVE)
```

Comment:
This is some absurd math going on here:

"If the user has at least 11/10ths the debt they owe, then the maximum they can pay is 10/9ths the debt they owe plus the minimum amount  we're allowed to leave remaining in a wallet to pay for gas costs, otherwise their balance"?